log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/EcsLayout.java

```diff
@@ -246,8 +263,19 @@ private void serializeMessage(StringBuilder builder, boolean gcFree, Message mes
 
     private static void serializeJsonMessage(StringBuilder builder, MultiformatMessage message) {
         final StringBuilder messageBuffer = EcsJsonSerializer.getMessageStringBuilder();
-        if (message instanceof MultiFormatStringBuilderFormattable) {
-            ((MultiFormatStringBuilderFormattable) message).formatTo(JSON_FORMAT, messageBuffer);
+        if (MULTI_FORMAT_STRING_BUILDER_FORMATTABLE.isInstance(message)) {
+            try {
+                FORMAT_TO.invoke(message, JSON_FORMAT, messageBuffer);
```

Comment:
An alternative to invoking this via reflection would be to have an isolated class calling `MultiFormatStringBuilderFormattable#formatTo` that's only loaded if the class is available.

The downside is that it wouldn't be possible to compile and test against the old version easily. However, we do want to test against the new version so that we can actually test this.

Reply:
> An alternative to invoking this via reflection would be to have an isolated class calling MultiFormatStringBuilderFormattable#formatTo that's only loaded if the class is available.

That's what I ended up with now.

> The downside is that it wouldn't be possible to compile and test against the old version easily. 

There's now a separate `log4j2-legacy-tests` module that tests against log4j 2.6 while the main module compiles against the most recent version (2.14.0)